Containerize the app for Local Deployment with Docker

fuguixing opened the pull request:
Why is it needed?

Previously, manually configuring a Silverstripe environment locally involved a cumbersome process of ensuring compatibility between various component versions. This PR alleviates this burden by providing a repeatable and reliable containerization approach.

The Dockerfile and docker-compose files handle all the necessary steps involved in setting up the environment. Developers can focus more on developing features and less on managing infrastructure, thanks to this PR!

fuguixing committed:
Add docker compose for rapid local deployment

leochenftw commented on Dockerfile:
```diff
@@ -0,0 +1,24 @@
+
+# Copy Vue.js dist
+RUN mkdir -p public/_resources/app/client/dist
+RUN cp -R app/client/dist/* /var/www/html/public/_resources/app/client/dist
```
can we use `ln -s`  instead?

leochenftw replied:
You can actually remove lines 23 and 24, and replace them with `RUN vendor/bin/sake dev/build flush=all`. Because line 20 should have created the directory, and the `sake dev/build` command will expose the `dist` directory automatically

leochenftw committed:
Update Dockerfile

Just updated how the dist directory was linked in `public`, and added the db build command

fuguixing committed:
Update Dockerfile

Switch to the non-root user to execute `composer vendor-expose`. Additionally, it includes a script in the entrypoint to ensure that the db is built after the MySQL service is ready.
